src/clj/puppetlabs/trapperkeeper/services/protocols/filesystem_watch_service.clj

```diff
@@ -6,9 +6,15 @@
   "Schema for an event on a file watched by this service."
   (schema/if #(= (:type %) :unknown)
     {:type (schema/eq :unknown)
-     :path (schema/pred nil?)}
+     :count schema/Int
+     :watched-path File
+     :changed-path (schema/pred nil?)
+     :full-path (schema/pred nil?)}
```

Comment:
... or `(schema/maybe File)`.

Also, what is `full-path`?  It's not obvious so it probably needs to be defined in the docstring.